sdk/search/Azure.Search.Documents/src/Generated/Models/AnswerResult.Serialization.cs

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+using System.Collections.Generic;
+using System.Text.Json;
+using Azure.Core;
+
+namespace Azure.Search.Documents.Models
+{
+    internal partial class AnswerResult
+    {
+        internal static AnswerResult DeserializeAnswerResult(JsonElement element)
+        {
+            Optional<double> score = default;
+            Optional<string> key = default;
+            Optional<string> text = default;
+            Optional<string> highlights = default;
+            IReadOnlyDictionary<string, object> additionalProperties = default;
+            Dictionary<string, object> additionalPropertiesDictionary = new Dictionary<string, object>();
+            foreach (var property in element.EnumerateObject())
+            {
+                if (property.NameEquals("score"))
+                {
+                    if (property.Value.ValueKind == JsonValueKind.Null)
+                    {
+                        property.ThrowNonNullablePropertyIsNull();
+                        continue;
+                    }
+                    score = property.Value.GetDouble();
+                    continue;
+                }
+                if (property.NameEquals("key"))
+                {
+                    key = property.Value.GetString();
+                    continue;
+                }
+                if (property.NameEquals("text"))
+                {
+                    text = property.Value.GetString();
+                    continue;
+                }
+                if (property.NameEquals("highlights"))
+                {
+                    if (property.Value.ValueKind == JsonValueKind.Null)
+                    {
+                        highlights = null;
+                        continue;
+                    }
+                    highlights = property.Value.GetString();
+                    continue;
+                }
+                additionalPropertiesDictionary.Add(property.Name, property.Value.GetObject());
+            }
+            additionalProperties = additionalPropertiesDictionary;
+            return new AnswerResult(Optional.ToNullable(score), key.Value, text.Value, highlights.Value, additionalProperties);
```

Comment:
I'm a bit confused here, why is `score` above considered non-nullable but here it appears to be turned into nullable?

Reply:
`AnswerResult.Score` is nullable double.
The code block at line 26-36 is verifying that if a property by the name 'score' is passed, the corresponding value is not null.
The ctor at line 59 is allowing a null to be passed if `score` was not set.